### DISCUSSION TITLE
How to set a milestone of a pull request?

### DISCUSSION THREAD

**Discussion by vadi2** [community/community#25724] [Actions]
<p>I’d like to set a milestone on a PR with -</p>
<pre><code class="lang-auto">curl -s --request PATCH \
  --url https://api.github.com/repos/Mudlet/Mudlet/issues/5916 \
  -H "Accept: application/vnd.github.v3+json" \
  --header 'authorization: token ghp_XXX' \
  --data '{"milestone": "4.16.0"}'
</code></pre>
<p>The token has <code>public_repo</code> permissions. The request fails with:</p>
<pre><code class="lang-json">{
  "message": "Validation Failed",
  "errors": [
    {
      "value": "4.16.0",
      "resource": "Issue",
      "field": "milestone",
      "code": "invalid"
    }
  ],
  "documentation_url": "https://docs.github.com/rest/reference/issues#update-an-issue"
}
</code></pre>
<p>Why is that and what does the error message mean?</p>

**Comment by vadi2** [comment, marked as answer]
<p>Milestone cannot be the title, but needs to be milestone ID as seen in <a href="https://github.com/Mudlet/Mudlet/milestone/" rel="noopener nofollow ugc">https://github.com/Mudlet/Mudlet/milestone/</a>&lt;id&gt;</p>

**Reply by scgomez** [reply]
Hi @vadi2, were you able to solve the problem to set the milestone to PR? I have tried from java but still no success, do you have any example how you did it from code?

**Reply by vadi2** [reply]
Yes I was, see [this](https://github.com/Mudlet/Mudlet/blob/development/.github/workflows/tag-pull-requests.yml) for an example workflow.

**Reply by scgomez** [reply]
Thanks for your reply, and sorry to bother you again. I'm executing curl from Postman, but it never assigns me the milestone, it's still milestone: null, is there something you can think of?

![image](https://user-images.githubusercontent.com/109813158/191133096-ec4df7af-a9f0-4801-9c8a-da8dcc4c8a6e.png)
![Captura de Pantalla 2022-09-19 a la(s) 5 55 42 p m](https://user-images.githubusercontent.com/109813158/191133630-6d781483-4bed-4081-b40e-5b6917773e5d.png)
![image](https://user-images.githubusercontent.com/109813158/191133185-478e142c-ef74-49ea-b627-74d40bb17f61.png)

**Reply by jamesjmtaylor-dexcom** [reply]
@scgomez In the url you posted you have `https://api.github.com/repose/OWNER/REPO/pulls/150`.  If you update it to `https://api.github.com/repose/OWNER/REPO/issues/150` it should work.
